### DISCUSSION TITLE
Capture Timestamp at which "yes/no" was selected or habit data was captured

### DISCUSSION THREAD

**Discussion by jaintj95** [iSoron/uhabits#1313] [Feature Requests]
Right now there's no way to see at "what time" did I perform a certain habit.  
Say the notification showed up at 2:00 PM but I only performed the habit at 4:16 PM (tapped "Yes" in notification).  
This data should be captured and then we should have the ability to view this data in habit statistics when we click on a certain day

**Comment by neekubee** [comment]
In addition, there would be a need to add the following:
(1) a time-frequency, a level deeper than the current day-frequency. Also, a manual time input if one indeed performed the task but somehow forgot to check, or there was an emergency. OR when checked, a follow-up opens to ask about time completion.
(2) a day-details, a level deeper than the current day-calendar. OR a time-axis in the history portion when chosen an interval of per-day.

**Reply by herrbrixter** [reply]
Your suggestion is too complicated. OP's idea is simpler. Let's start with capturing "updated time" before considering complex features.

**Comment by herrbrixter** [comment]
I want the timestamp displayed in the Yes or No popup like below. Here I clicked Yes at 9:34 AM:

![Screenshot_20251206_214008_Habits~(1)](https://github.com/user-attachments/assets/432de52c-32ce-489c-8fe5-985268b41482)

**Comment by herrbrixter** [comment]
Capturing timestamps is also useful for measureable habits. For example, drinking 8 glasses of water per day.

4 glasses at 11:27AM

At 5PM, I know that I drank 4 glasses already but I haven't drank for 5.5 hours since then.
